components/org.wso2.carbon.identity.discovery/src/main/java/org/wso2/carbon/identity/discovery/builders/ProviderConfigBuilder.java

```diff
@@ -118,16 +118,33 @@ public OIDProviderConfigResponse buildOIDProviderConfig(OIDProviderRequest reque
         providerConfig.setEndSessionEndpoint(buildServiceUrl(IdentityConstants.OAuth.LOGOUT,
                 IdentityUtil.getProperty(IdentityConstants.OAuth.OIDC_LOGOUT_EP_URL)));
 
-        try {
-            providerConfig.setUserinfoSigningAlgValuesSupported(new String[] {
-                    OAuth2Util.mapSignatureAlgorithmForJWSAlgorithm(
-                            OAuthServerConfiguration.getInstance().getUserInfoJWTSignatureAlgorithm()).getName()
-            });
-        } catch (IdentityOAuth2Exception e) {
-            throw new ServerConfigurationException("Unsupported signature algorithm configured.", e);
+        List<String> supportedUserinfoSigningAlgValues = OAuthServerConfiguration.getInstance()
+                .getUserInfoJWTSignatureAlgorithms();
+        String userInfoJWTSignatureAlgorithm =  OAuthServerConfiguration.getInstance()
+                .getUserInfoJWTSignatureAlgorithm();
+        if (!supportedUserinfoSigningAlgValues.contains(userInfoJWTSignatureAlgorithm)) {
+            supportedUserinfoSigningAlgValues.add(userInfoJWTSignatureAlgorithm);
```

Comment:
Since you are modifying the object in the OAuthServerConfiguration, there is no point in doing it again and again

Reply:
The object was modified in order to merge the values obtained from the existing config (UserInfoJWTSignatureAlgorithm) and the new config which supports multiple algorithms. Added a comment to the condition to give clarity to the context.